website_sale_charge_payment_fee/controllers/website_sale.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo.http import request
+from odoo import http
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class WebsiteSaleFee(WebsiteSale):
+
+    @http.route()
```

Comment:
Shouldn't the route be `@http.route(['/shop/payment'], type='http', auth="public", website=True)` ?

Reply:
If the decorator was declared in the parent method, you don't need to repeat things.

Reply:
Thanks, didn't know that, I usually keep repeating decorators in every overriding method.

Reply:
Yes, you need the decorator; just not the arguments.